Sources/Paywall/Paywall/Paywall View Controller/WebEventHandler.swift

```diff
@@ -0,0 +1,233 @@
+    )
+
+    guard let paywallResponse = delegate?.paywallResponse else {
+      return
+    }
+
+    switch event {
+    case .templateParamsAndUserAttributes:
+      templateParams(from: paywallResponse)
+    case .onReady:
+      didLoadWebView(from: paywallResponse)
+    case .close:
+      hapticFeedback()
+      delegate?.eventDidOccur(.closed)
+    case .openUrl(let url):
+      openUrl(url)
+    case .openDeepLink(let url):
+      openDeepLink(url)
+    case .restore:
+      restorePurchases()
+    case .purchase(product: let type):
+      purchaseProduct(
+        withType: type,
+        from: paywallResponse
+      )
+    case .custom(data: let customEvent):
+      handleCustomEvent(customEvent)
+    }
+  }
+
+  private func templateParams(from paywallResponse: PaywallResponse) {
+    let params = paywallResponse.getBase64EventsString(
+      params: delegate?.eventData?.parameters
+    )
+    let scriptSrc = """
+      window.paywall.accept64('\(params)');
+    """
+    delegate?.webView.evaluateJavaScript(scriptSrc) { _, error in
+      if let error = error {
+        Logger.debug(
+          logLevel: .error,
+          scope: .paywallViewController,
+          message: "Error Evaluating JS",
+          info: ["message": scriptSrc],
+          error: error
+        )
+      }
+    }
+
+    Logger.debug(
+      logLevel: .debug,
+      scope: .paywallViewController,
+      message: "Posting Message",
+      info: ["message": scriptSrc],
+      error: nil
+    )
+  }
+
+  private func didLoadWebView(from paywallResponse: PaywallResponse) {
+    if let paywallInfo = delegate?.paywallInfo {
+      if paywallResponse.webViewLoadCompleteTime == nil {
+        delegate?.paywallResponse?.webViewLoadCompleteTime = Date()
+      }
+
+      Paywall.track(.paywallWebviewLoadComplete(paywallInfo: paywallInfo))
+    }
+
+    let params = paywallResponse.getBase64EventsString(
+      params: delegate?.eventData?.parameters
+    )
+    let jsEvent = paywallResponse.paywalljsEvent
+    let scriptSrc = """
+      window.paywall.accept64('\(params)');
+      window.paywall.accept64('\(jsEvent)');
+    """
+
+    Logger.debug(
+      logLevel: .debug,
+      scope: .paywallViewController,
+      message: "Posting Message",
+      info: ["message": scriptSrc],
+      error: nil
+    )
+
+    delegate?.webView.evaluateJavaScript(scriptSrc) { [weak self] _, error in
+      if let error = error {
+        Logger.debug(
+          logLevel: .error,
+          scope: .paywallViewController,
+          message: "Error Evaluating JS",
+          info: ["message": scriptSrc],
+          error: error
+        )
+      }
+      self?.delegate?.loadingState = .ready
+    }
+
+    // block selection
+    // swiftlint:disable:next line_length
+    let selectionString = "var css = '*{-webkit-touch-callout:none;-webkit-user-select:none} .w-webflow-badge { display: none !important; }'; "
+      + "var head = document.head || document.getElementsByTagName('head')[0]; "
```

Comment:
The inline warnings are so dope!!

Reply:
The strange thing is SwiftLint warns there's no error on this line when running in GitHub, but gives a warning when running in xcode